Description: Revert "Enforcing static invariants to prevent failures
 arising from layout"
 This reverts commit f4e573399592c89b9a35451cb2f39fc6a6e0c6f2.
Forwarded: not-needed
--- a/packages/modules/adb/types.h
+++ b/packages/modules/adb/types.h
@@ -98,7 +97,6 @@ struct Block {
     const char& operator[](size_t idx) const { return data()[idx]; }
 
     bool operator==(const Block& rhs) const {
-        static_assert(std::is_standard_layout<decltype(data())>());
         return size() == rhs.size() && memcmp(data(), rhs.data(), size()) == 0;
     }
 
@@ -232,7 +230,6 @@ struct IOVector {
 
         size_t offset = 0;
         iterate_blocks([&offset, &result](const char* data, size_t len) {
-            static_assert(std::is_standard_layout<decltype(result)>());
             memcpy(&result[offset], data, len);
             offset += len;
         });
